zipline/gens/brokers/ib_broker.py

```diff
@@ -404,8 +404,9 @@ def positions(self):
                 continue
             z_position.amount = int(ib_position.position)
             z_position.cost_basis = float(ib_position.average_cost)
-            z_position.last_sale_price = None  # TODO(tibor): Fill from state
-            z_position.last_sale_date = None  # TODO(tibor): Fill from state
+            z_position.last_sale_price = float(
+                self._tws.bars[symbol].last_trade_price.iloc[-1])
```

Comment:
It might happen that `self._tws.bars[symbol]` is unpopulated at the beginning of the session. Currently we are subscribing the market data on the first `data.current()` call. If the algo tries to access the position before making a call this statement will yield a KeyError.

Could you please add a check for symbol existence in `bars`?
If the symbol does not exists we could return NaN or None (not sure which is better).

The same applies for `last_sale_date` below.